tests/support.py

```diff
@@ -0,0 +1,29 @@
+try:
+    import __builtin__ as builtins
+except ImportError:
+    # Python 3...
+    import builtins
+
+"""
+This module exists to override the open function and log
+what files get opened.
+"""
+
+opened_files = []
+
+
+def clear_buffer():
+    global opened_files
+    opened_files = []
+
+
+oldopen = builtins.open
+
+
+def newopen(*args):
+    global opened_files
+    fh = oldopen(*args)
+    opened_files.append(fh.name)
+    return fh
+
+builtins.open = newopen
```

Comment:
It's slightly more involved but this could be achieved without the global variable or the unbound wrapping of a builtin with a couple of closures in a pytest fixture.

```python
@pytest.fixture(autouse=True)
def proxy_open(request):
    old_open = builtins.open
    opened_files = []

    def new_open(*args, **kwargs):
        fh = old_open(*args, **kwargs)
        opened_files.append(fh.name)
        return fh

    builtins.open = new_open

    def undo():
        builtins.open = old_open

    request.addfinalizer(undo)

    return opened_files
```